src/metabase/python_runner/api.clj

```diff
@@ -63,33 +63,72 @@
+        work-dir (str mount-path "/" work-dir-name)
+        work-dir-file (io/file work-dir)]
+
+    ;; Ensure mount base path exists
+    (.mkdirs (io/file mount-path))
+    ;; Create working directory
+    (.mkdirs work-dir-file)
+
+    (try
+      (let [server-url (transforms.settings/python-execution-server-url)
+            api-key (transforms.settings/python-runner-api-key)
+            metabase-url (transforms.settings/python-runner-callback-base-url)
+
+            response (http/post (str server-url "/execute")
+                                {:content-type     :json
+                                 :accept           :json
+                                 :body             (json/encode {:code          code
+                                                                 :working_dir   work-dir
+                                                                 :timeout       30
+                                                                 :metabase_url  metabase-url
+                                                                 :api_key       api-key
+                                                                 :table_mapping table-name->id})
+                                 :throw-exceptions false
+                                 :as               :json})
+
+            result  (:body response)
+            ;; TODO look into why some tests return json and others strings
+            result (if (string? result) (json/decode result keyword) result)]
+
+        (try
+          (if (and (= 200 (:status response))
+                   (zero? (:exit_code result)))
+            ;; Success - read the output CSV if it exists
+            (let [output-path (:output_file result)]
+              (if (and output-path (.exists (io/file output-path)))
+                {:output (slurp output-path)
+                 :stdout (safe-slurp (:stdout_file result))
+                 :stderr (safe-slurp (:stderr_file result))}
+                {:status 500
+                 :body   {:error  "Transform did not produce output CSV"
+                          :stdout (safe-slurp (:stdout_file result))
+                          :stderr (safe-slurp (:stderr_file result))}}))
+            ;; Error from execution server
+            {:status 500
+             :body
+             {:error     (or (:error result) "Execution failed")
+              :exit-code (:exit_code result)
+              :timeout   (:timeout result)
+              :stdout    (safe-slurp (:stdout_file result))
+              :stderr    (safe-slurp (:stderr_file result))}})
+          (finally
+            ;; Clean up working directory after use
+            (try
+              (when (.exists work-dir-file)
+                ;; Delete directory and all contents
+                (doseq [file (reverse (file-seq work-dir-file))]
+                  (.delete ^File file)))
+              (catch Exception _)))))
```

Comment:
Still need complementary garbage pruning *before* each execution, in case this fails